jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java

```diff
@@ -782,16 +789,34 @@ private Response requestAuthorization(
     }
 
     @Nullable
-    private ResponseMode extractResponseMode(String request) {
-        final Jwt jwt = Jwt.parseSilently(request);
-        if (jwt == null) {
+    private Jwe extractJwe(String request, Client client) {
```

Comment:
This is not going to work, we can flip it from Jwt to Jwe and then from Jwe to Jwt depending on use case. It will never ends. Instead we should try both. For example: 
```
private JsonWebResponse extractResponseMode(String request) {
     String[] parts = request.split("\\.");
     if (parts.length == 5) {
         // parse jwe
         ...
         return jwe;
     }
     
     // parse jwt
     ...
     return jwt;
}
```

Reply:
Not sure, as in case of JARM we only get encrypted request so should we include the other case too?

Reply:
@HemantKMehta  Why do you think so?
In spec, section 4.2 I see this:
`The JWT is either signed, or signed and encrypted. If the JWT is both signed and encrypted, the JSON document will be signed then encrypted, with the result being a Nested JWT, as defined in [RFC7519].`

Reply:
but it may return null for other cases so when using it I need to add as condition to check for null value 
                Jwe jwe = extractJwe(request, client);
                 if (jwe !=null)  {
                     ... 
                     ...
                 }

Reply:
Oh.. Yes its both signed and encrypted. 
I am changing it as you suggested, Thanks

Reply:
> but it may return null for other cases so when using it I need to add as condition to check for null value

Did you really take a look to example I posted above? How can you get null if before you checked parts?

Reply:
> > but it may return null for other cases so when using it I need to add as condition to check for null value
> 
> Did you really take a look to example I posted above? How can you get null if before you checked parts?
Actually this message was for my code with extractJwe method not for the example you have given. 
I am updating it as you suggested.